src/common/reducers/snap-builds.js

```diff
@@ -0,0 +1,31 @@
+import * as ActionTypes from '../actions/snap-builds';
+import { snapBuildFromAPI } from '../helpers/snap-builds';
+
+export function snapBuilds(state = {
+  isFetching: false,
+  builds: [],
+  error: false
+}, action) {
+  switch(action.type) {
+    case ActionTypes.FETCH_BUILDS:
+      return {
+        ...state,
+        isFetching: true
+      };
+    case ActionTypes.FETCH_BUILDS_SUCCESS:
+      return {
+        ...state,
+        isFetching: false,
+        builds: action.payload.map(snapBuildFromAPI),
```

Comment:
Most of the 'logic' of transforming of API returned entry into values in the store is done in `snapBuildFromAPI` helper.